### ISSUE TITLE
Global plug-in exception problem  ！！！

### ISSUE THREAD

**Issue by githubxubin** [apache/apisix#11168]
### Description

I need to use the jwt-auth plugin and I install the documentation for the related configuration
First up:
I have enabled the jwt-auth plugin in the global plugin
Then, I created a public-api route according to the documentation in order to get the token:
```
curl http://127.0.0.1:9180/apisix/admin/routes/jas \
-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "uri": "/apisix/plugin/jwt/sign",
    "plugins": {
        "public-api": {}
    }
}'
```
Result: public-api this route can not be accessed, prompting the need for token, this is not a serious bug,public-api should not conduct token verification ah

<img width="747" alt="image" src="https://github.com/apache/apisix/assets/39228009/0d798f06-be05-4222-b930-fe03930dd042">

apisix version:3.6.0-centos
dashboard:3.0
etcd:3.4.15

**Comment by kayx23** [comment]
I cannot recreate. Is your usage similar to the first two sections here: https://docs.api7.ai/hub/jwt-auth#examples

**Comment by githubxubin** [comment]
> /apisix/plugin/jwt/sign

Yes, the JWt-Auth plug-in is enabled globally, the route "/apisix/plugin/jwt/sign" cannot be obtained, and a message is displayed indicating that the JWT token is missing. Is this normal? Does the route "/apisix/plugin/jwt/sign" need a token？

**Comment by kayx23** [comment]
The error you encountered is unexpected. Could you make sure you have no other conflicting routes (i.e. matching the same path)
